src/object_data.hpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file
+ * @brief     This file contains a class that represents the data that is needed to predict positions of an object.
+ * @author    Daniel van vliet
+ * @license   See LICENSE
+ */
+
+#ifndef OBJECTDATA_HPP
+#define OBJECTDATA_HPP
+#include "vector3d.hpp"
+#include <array>
+
+class ObjectData {
+  private:
+    std::array<Vector3D, 10> positions;
```

Comment:
You may want to make the ObjectData object a template. In this way, users are able to provide their own desired array sizes.